lib/Dialect/TTNN/Transforms/TTNNWorkarounds.cpp

```diff
@@ -212,20 +249,31 @@ static bool workaroundOutputOperand(
                   outputWorkaroundResult.targetTensorBufferTypeResult.first)
               .withMemoryLayout(rewriter.getContext(), outputMemLayoutAttr));
 
+  // Preserve the previous data type of the DPS result.
+  DataType previousOutputDataType =
+      getLayoutAttrFromValue(opResult).getDataType();
```

Comment:
nit: since this function works with function results maybe some diffent different name is better then workaroundOutputOperand?